sw/device/tests/sysrst_ctrl_inputs_test.c

```diff
@@ -0,0 +1,154 @@
+    kTopEarlgreyPinmuxInselIor4, kTopEarlgreyPinmuxInselIor5,
+    kTopEarlgreyPinmuxInselIor6, kTopEarlgreyPinmuxInselIor7,
+};
+
+static const dif_sysrst_ctrl_pin_t kSysrstCtrlInputs[] = {
+    kDifSysrstCtrlPinKey0In,           kDifSysrstCtrlPinKey1In,
+    kDifSysrstCtrlPinKey2In,           kDifSysrstCtrlPinPowerButtonIn,
+    kDifSysrstCtrlPinAcPowerPresentIn, kDifSysrstCtrlPinLidOpenIn,
+    kDifSysrstCtrlPinEcResetInOut,     kDifSysrstCtrlPinFlashWriteProtectInOut,
+};
+
+static uint8_t read_input_pins(void) {
+  bool input_value;
+  uint8_t inputs = 0;
+  for (int i = 0; i < kOutputNumPads; ++i) {
+    CHECK_DIF_OK(dif_sysrst_ctrl_input_pin_read(
+        &sysrst_ctrl, kSysrstCtrlInputs[i], &input_value));
+    inputs |= input_value << i;
+  }
+  return inputs;
+}
+
+void setup_dio_pins(dif_pinmux_t *pinmux, dif_sysrst_ctrl_t *sysrst_ctrl) {
+  // Make sure the the DIO pins EC reset and flash WP are configured correctly.
+  // The documentation says that they should be configured as open drain but on
+  // FPGA targets, only virtual open drain is supported.
+  dif_pinmux_pad_attr_t out_attr;
+  dif_pinmux_pad_attr_t in_attr = {
+      .slew_rate = 0,
+      .drive_strength = 0,
+      // FIXME FPGA requires virtual open drain, should be open drain on real
+      // chip?
+      .flags = kDifPinmuxPadAttrVirtualOpenDrain,
+  };
+  CHECK_DIF_OK(dif_pinmux_pad_write_attrs(
+      pinmux, kTopEarlgreyDirectPadsSysrstCtrlAonEcRstL, kDifPinmuxPadKindDio,
+      in_attr, &out_attr));
+  CHECK_DIF_OK(dif_pinmux_pad_write_attrs(
+      pinmux, kTopEarlgreyDirectPadsSysrstCtrlAonFlashWpL, kDifPinmuxPadKindDio,
+      in_attr, &out_attr));
+  // We also need to disable the output override mecanism (i.e. "release the
+  // pin").
+  dif_sysrst_ctrl_pin_config_t cfg = {
+      .enabled = false,
+      .allow_zero = true,
+      .allow_one = true,
+      .override_value = false,
+  };
+  CHECK_DIF_OK(dif_sysrst_ctrl_output_pin_override_configure(
+      sysrst_ctrl, kDifSysrstCtrlPinEcResetInOut, cfg));
+  CHECK_DIF_OK(dif_sysrst_ctrl_output_pin_override_configure(
+      sysrst_ctrl, kDifSysrstCtrlPinFlashWriteProtectInOut, cfg));
+  // Disable the EC reset debounce timer because it could interfere with
+  // reading the pins.
+  mmio_region_write32(sysrst_ctrl->base_addr, SYSRST_CTRL_EC_RST_CTL_REG_OFFSET,
+                      0);
+}
+
+bool test_main(void) {
+  CHECK_DIF_OK(dif_sysrst_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_SYSRST_CTRL_AON_BASE_ADDR),
+      &sysrst_ctrl));
+
+  dif_pinmux_t pinmux;
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+
+  /* On real devices, we also need to configure the DIO pins */
+  if (kDeviceType != kDeviceSimDV)
+    setup_dio_pins(&pinmux, &sysrst_ctrl);
+  const dif_pinmux_index_t *kInputPads =
+      kDeviceType == kDeviceSimDV ? kInputPadsDV : kInputPadsReal;
+  for (int i = 0; i < kOutputNunMioPads; ++i) {
```

Comment:
The size of `kPeripheralInputs` is 8, while that of `kInputPads` is only 6, and `kOutputNunMioPads` is 6, so the last 2 peripheral inputs are not set, and that includes flash protect.

Reply:
Thanks @matutem  I don't think it explains the problem: the reason why kOutputNunMioPads is 6 is because the flash protect and ec reset pins are direct I/O pins (EC reset is always IOR8 and flash WP is always IOR9) so they are not configured by dif_pinmux_input_select. At least that's my understanding of the pinmux.
@a-will Could you have a look at this?

Reply:
@pamaury is correct. The pinmux only needs to be configured for the 6 pads (as it was in the sim_dv test). IOR8 / `ec_rst_l` and IOR9 / `flash_wp_l` were converted to be DIOs, and they were annoyingly never renamed:

https://github.com/lowRISC/opentitan/blob/755180702e6e410ad895bf48c06bbce571670502/hw/top_earlgrey/data/top_earlgrey.hjson#L1276-L1277